js/interface.js

```diff
@@ -606,9 +608,11 @@ function resetPagination() {
   pageSize = +$('#page-size').val() || undefined;
 }
 
-function fetchCurrentDataSourceEntries() {
-  resetPagination();
-
+function fetchCurrentDataSourceEntries(paginationReset = true) {
```

Comment:
1. Use an `options` object so it's clearer in the caller context what the parameter is. `fetchCurrentDataSourceEntries(false)` isn't very obvious when reading what the `false` refers to.
2. `resetPagination` is a more suitable property name as it's an action (verb first) indicating it'll affect whether something is actioned or not

```suggestion
function fetchCurrentDataSourceEntries(options = {}) {
  if (options.paginationReset !== false)
```